### DISCUSSION TITLE
node:events:368 - unhandled error - M1 Max MacBook

### DISCUSSION THREAD

**Discussion by chris-hinds** [vercel/next.js#32202] [Help] [answered]
Hi,

For a little while now I have been having this issue when running `yarn dev` on an existing nextJS project.

```
info  - Using external babel configuration from /Users/chris/workspace/XXX/XXX/babel.config.js
node:events:368
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:211:20)
Emitted 'error' event on ClientRequest instance at:
    at TLSSocket.socketErrorListener (node:_http_client:447:9)
    at TLSSocket.emit (node:events:390:28)
    at TLSSocket.emit (node:domain:475:12)
    at emitErrorNT (node:internal/streams/destroy:164:8)
    at emitErrorCloseNT (node:internal/streams/destroy:129:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -54,
  code: 'ECONNRESET',
  syscall: 'read'
}

Node.js v17.2.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I have tried running this project across multiple versions of node. And nearly always face the same issue.

I cannot share the project as it is private. However other colleges not running on Apple Silicon are not having this issue.

**Comment by asobirov** [comment]
Have you tried to use yarn classic? I think I used to have the same issue with yarn v3

**Reply by chris-hinds** [reply]
I'm already using yarn v1 ☹️

**Comment by janmisek1** [comment]
Same here, but we not use next.js. Only webpack...

**Comment by chris-hinds** [comment, marked as answer]
It seems that moving to the very latest release of Node 17 solved my issue.

**Reply by janmisek1** [reply]
Not work for me.

**Reply by chris-hinds** [reply]
@janmisek1 hope are you running your terminal, natively with ARM or via rosetta (emulated x86) this may make a difference as it will install ARM or x86 binaries for node.

I am now running my terminal natively and using the ARM build for node.

**Reply by hamidash** [reply]
@chris-hinds I have a Intel Core i5 , MacPro, and just checked and both my Native terminal and VSCode terminal give me `i386` when I run `arch` command. But I am still facing the same issue? Is there any solution for intel chip macs?
